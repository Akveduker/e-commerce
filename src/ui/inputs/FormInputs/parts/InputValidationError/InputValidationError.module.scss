@import "../../../../../styles/_mixins.scss";
.error {
    @include authorizationError;
}
