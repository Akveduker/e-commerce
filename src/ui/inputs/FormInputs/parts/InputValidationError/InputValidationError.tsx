import React, { FC } from 'react';
import { STATUS_ERROR } from '../../../../../data/status/status';
import { Status } from '../../../../../types/categories/categories';

import s from './InputValidationError.module.scss'
interface InputValidationErrorProps {
    status: Status;
    errorBody: string;
}
const InputValidationError: FC<InputValidationErrorProps> = ({ status, errorBody }) => {
    return (
        <div>
            {status === STATUS_ERROR && <p className={s['error']}>{errorBody}</p>}
        </div>
    );
};

export default InputValidationError;