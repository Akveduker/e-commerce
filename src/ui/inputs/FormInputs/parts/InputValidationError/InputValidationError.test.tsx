import { render, screen } from "@testing-library/react"
import InputValidationError from "./InputValidationError"

describe('InputValidationError', () => {
    test('Элемент не рендерится, снапшот', () => {
        const { container } = render(<InputValidationError errorBody="error" status="idle" />)
        expect(screen.queryByText('error')).toBeNull()
        expect(container).toMatchSnapshot()
    })
    test('Элемент рендерится, снапшот', () => {
        const { container } = render(<InputValidationError errorBody="error" status="error" />)
        expect(screen.queryByText('error')).toBeInTheDocument()
        expect(container).toMatchSnapshot()
    })
})