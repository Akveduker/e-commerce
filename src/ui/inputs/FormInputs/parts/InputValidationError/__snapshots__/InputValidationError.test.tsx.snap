// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputValidationError Элемент не рендерится, снапшот 1`] = `
<div>
  <div />
</div>
`;

exports[`InputValidationError Элемент рендерится, снапшот 1`] = `
<div>
  <div>
    <p
      class="error"
    >
      error
    </p>
  </div>
</div>
`;
